core-metadata/src/main/java/org/apache/kylin/metadata/model/FunctionDesc.java

```diff
@@ -146,7 +146,11 @@ public boolean needRewriteField() {
 
     public String getRewriteFieldName() {
         if (isCount()) {
-            return "_KY_" + "COUNT__"; // ignores parameter, count(*), count(1), count(col) are all the same
+            if (parameter == null || parameter.getType().equals("constant")) {
```

Comment:
Use "org.apache.kylin.metadata.model.ParameterDesc#isColumnType" method will be more clear.

Reply:
done